partiql-ast/src/main/java/org/partiql/ast/v1/Identifier.java

```diff
@@ -0,0 +1,80 @@
+package org.partiql.ast.v1;
+
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * TODO docs, equals, hashcode
+ */
+public abstract class Identifier extends AstNode {
+    @Override
+    public <R, C> R accept(@NotNull AstVisitor<R, C> visitor, C ctx) {
+        if (this instanceof Symbol) {
+            return visitor.visitIdentifierSymbol((Symbol) this, ctx);
+        } else if (this instanceof Qualified) {
+            return visitor.visitIdentifierQualified((Qualified) this, ctx);
+        } else {
+            throw new IllegalStateException("Unexpected value: " + this);
+        }
+    }
+
+    /**
+     * TODO docs, equals, hashcode
+     */
+    public static class Symbol extends Identifier {
```

Comment:
SQL:1999 also has `<local or schema qualified name>`, though, it doesn't fit our use-case particularly well, since we don't have the same two-level catalog structure. Could be of use to adopt this name. `Identifier` and `IdentifierChain`.

Reply:
Thanks for the suggestion. I've been confused at times by the previous names, `Symbol` and `Qualified`, so open to changing them.

I might need some clarity on the suggestion. Would `Identifier` and `IdentifierChain` be within some abstract class or be separate `AstNode`s entirely? I believe you're referring to separate `AstNode`s without some overarching abstract class but just checking before I refactor further.

Reply:
Changed to use `Identifier` and `IdentifierChain` as separate nodes.